server/src/model.ts

```diff
@@ -907,6 +931,89 @@ async function createDocumentVersion(docId: Uint8Array): Promise<{
   return docVersion;
 }
 
+function filterViewableActivity(loggedInUserId: Uint8Array, isAdmin: boolean) {
+  // For an activity to be viewable, one of these conditions must be true:
+  // 1. You are the owner
+  // 2. The activity is public
+  // 3. The activity is shared with you
+  // 4. You are an admin and the activity is in the library
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const visibilityOptions: any[] = [
```

Comment:
I don't know a better way to get the type than to be verbose and repetitive... It's at least better than `any[]`.
```
  const visibilityOptions: (
    | { ownerId: Uint8Array }
    | { isPublic: boolean }
    | { sharedWith: { some: { userId: Uint8Array } } }
    | { owner: { isLibrary: boolean } }
  )[] 
```